File: ios-sdk/FlagshipFeatureFlags/Example/FlagshipFeatureFlags/ConfigHelper.swift
//
//  ConfigHelper.swift
//  FlagshipFeatureFlags
//
//  Created on 2025.
//

import Foundation

struct ConfigHelper {
    struct Config: Codable {
        let baseUrl: String
        let flagshipApiKey: String
    }
    
    static func loadConfig() -> Config {
        guard let url = Bundle.main.url(forResource: "config", withExtension: "json"),
              let data = try? Data(contentsOf: url),
              let config = try? JSONDecoder().decode(Config.self, from: data) else {
            // Fallback to default values if config file is not found
            return Config(
                baseUrl: "https://dream11.flagshiphq.io",
                flagshipApiKey: "XPCkqeT39sOH5WIUUvljJ11QhRjw6QeE"
            )
        }
        return config
    }
}

File: android-sdk/app/src/main/java/com/flagship/androidsdk/ConfigHelper.kt
package com.flagship.androidsdk

import android.content.Context
import org.json.JSONObject
import java.io.IOException

object ConfigHelper {
    data class Config(
        val baseUrl: String,
        val flagshipApiKey: String
    )

    fun loadConfig(context: Context): Config {
        return try {
            val jsonString = context.assets.open("config.json").bufferedReader().use { it.readText() }
            val jsonObject = JSONObject(jsonString)
            Config(
                baseUrl = jsonObject.getString("baseUrl"),
                flagshipApiKey = jsonObject.getString("flagshipApiKey")
            )
        } catch (e: IOException) {
            // Fallback to default values if config file is not found
            Config(
                baseUrl = "https://dream11.flagshiphq.io",
                flagshipApiKey = "XPCkqeT39sOH5WIUUvljJ11QhRjw6QeE"
            )
        }
    }
}

